perllib/FixMyStreet/Cobrand/UKCouncils.pm

```diff
@@ -171,17 +171,88 @@ sub area_check {
 
     my $councils = $params->{all_areas};
 
-    # The majority of cobrands only cover a single area, but e.g. Northamptonshire
-    # covers multiple so we need to handle that situation.
-    my $council_area_ids = $self->council_area_id;
-    $council_area_ids = [ $council_area_ids ] unless ref $council_area_ids eq 'ARRAY';
-    foreach (@$council_area_ids) {
-        return 1 if defined $councils->{$_};
-    }
+    return 1 if $self->responsible_for_areas($councils);
 
     return ( 0, $self->area_check_error_message($params, $context) );
 }
 
+=head2 responsible_for_areas
+
+Tests to see if the cobrand is responsible for a location. This is done through
+checking if it has the area set in council_area_id and also checking if the
+coverage is limited only to an overlapping asset by testing the hook
+report_is_on_cobrand_asset
+
+=cut
+
+sub responsible_for_areas {
+    my ($self, $councils) = @_;
+
+    if ($self->{c}->cobrand->can('report_is_on_cobrand_asset')) {
+        if (scalar(%$councils) == 1 && (values %$councils)[0]->{name} eq $self->council_name) {
```

Comment:
This would presumably have to do something different in two-tier areas, not sure worth doing without a concrete need, but perhaps worth a note. Not sure we want to rely on names not changing in MapIt ever here, either, even if unlikely - could it go on the first ID in $self->council_area_id perhaps, then more matches the else case.